routers/docs.py

```diff
@@ -1,24 +1,100 @@
 import http.client
 import logging
 
-from fastapi import APIRouter, Depends, HTTPException
+from fastapi import APIRouter, BackgroundTasks, Depends, HTTPException
+
+from schemas.api import GenerateFileDocsRequest, GenerateFileDocsResponse, GetFileDocsResponse, DocsStatusEnum
+from schemas.database import FirestoreDocumentationCreateModel, FirestoreDocumentationUpdateModel
 
-from schemas.generated_doc import GenerateFileDocRequest, GeneratedDocResponse
 from services.documentation_service import DocumentationService, get_documentation_service
+from services.clients.firebase_client import FirebaseClient, get_firebase_client
 
 router = APIRouter()
 
+# background tasks for generating the documentation
+async def generate_documentation(
+    documentation_service: DocumentationService,
+    firebase_client: FirebaseClient,
+    firebase_file_id: str,
+    github_url: str, 
+):
+    generated_docs = await documentation_service.generate_doc_for_github_file(github_url)
+
+    blob_url = firebase_client.get_blob_url(firebase_file_id)
+
+    firebase_client.add_blob(blob_url, generated_docs.content)
+    
+    firebase_client.update_document(
+        FirebaseClient.TEST_COLLECTION, 
+        firebase_file_id, 
+        FirestoreDocumentationUpdateModel
+        (
+            bucket_url=blob_url,
+            status=DocsStatusEnum.COMPLETED
+        ).model_dump()
+    )
 
-@router.post("/file-doc")
-async def generate_file_doc(
-        generate_file_doc_request: GenerateFileDocRequest,
-        documentation_service: DocumentationService = Depends(get_documentation_service)
-) -> GeneratedDocResponse:
+
+@router.post("/file-docs")
```

Comment:
We should add a status code of 202 (Accepted) here:
`@router.post("/file-doc", status_code=http.client.ACCEPTED)`

Meaning we accepted the request and are processing it in the background.